spec/models/move_spec.rb

```diff
@@ -219,8 +219,28 @@
     it { is_expected.to validate_presence_of(:reference) }
   end
 
+  describe '#nomis_event_id' do
+    subject(:move) { build :move }
+
+    context 'when nomis_event_ids array is not present' do
+      it 'returns nil' do
+        expect(move.nomis_event_id).to be_nil
+      end
+    end
+
+    context 'when nomis_event_ids array is present' do
+      before do
+        move.nomis_event_ids = [123_456, 654_321]
+      end
+
+      it 'returns the most recent nomis_event_id' do
+        expect(move.nomis_event_id).to eq(654_321)
+      end
+    end
+  end
+
   describe '#nomis_event_id=' do
-    subject(:move) { create :move }
+    subject(:move) { build :move }
```

Comment:
🎉

Reply:
Every optimisation helps... but this is soon to be removed completely 💣

Reply:
👍